plugins/fusion/fusion.py

```diff
@@ -364,7 +373,8 @@ def check_coins(self):
     def clear_coins(self):
         """ Clear the inputs list and release frozen coins. """
         for wallet, (coins, mytxids, checked_txids) in self.source_wallet_info.items():
-            wallet.set_frozen_coin_state(coins, False)
+            with wallet.lock:
```

Comment:
Why lock here? ` set_frozen_coin_state` already takes this lock when it is performing its operations.

Reply:
d'oh, you're right -- I was looking at the wrong one (`set_frozen_state`)!

Reply:
Ok.. waiting for your unfix to the non-fixing-anything fix.